src/main/java/uk/gov/hmcts/ccd/security/filters/RoleLoggingFilter.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.ccd.security.filters;
+
+import com.microsoft.applicationinsights.telemetry.SeverityLevel;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import uk.gov.hmcts.ccd.AppInsights;
+import uk.gov.hmcts.ccd.data.SecurityUtils;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+@Component
+@ConditionalOnProperty(value = "role.logging.enabled", havingValue = "true")
+@Slf4j
+public class RoleLoggingFilter extends OncePerRequestFilter {
+
+    private static final String LOG_MESSAGE_TEMPLATE = "[ROLE LOG] Attempting to serve request %s %s for user with IDAM roles %s";
+
+    private final AppInsights appInsights;
+    private final SecurityUtils securityUtils;
+    private final Pattern pathPattern;
+
+    @Autowired
+    public RoleLoggingFilter(AppInsights appInsights, SecurityUtils securityUtils, @Value("${role.logging.path.regex}") String pathRegex) {
+        this.appInsights = appInsights;
+        this.securityUtils = securityUtils;
+        this.pathPattern = Pattern.compile(pathRegex);
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+        throws ServletException, IOException {
+        String message = buildLogMessage(request, securityUtils.getUserInfo());
```

Comment:
i guess it won't log in case of auth failures. Is that OK?

Reply:
That's right, on the 3A was decided we only want it for authorised requests 👍

Reply:
Well, I would suggest to log all requests no matter authorisation passes or not. We had some cases to diagnose with role-based authorisation failures and we may have more.

Reply:
Apologies @MSancaktutar, in my haste I put authorised when I meant authenticated.. If a request is unauthorised (e.g. user requests case they don't have access to) it DOES still log!

Reply:
Well, in fact, what I meant was... A user's roles not being sufficient to access and operation would be an authorisation failure. And even in that case, we should log the roles.